apps/test-suite/screens/TestScreen.js

```diff
@@ -0,0 +1,371 @@
+'use strict';
+
+import React from 'react';
+import {
+  Dimensions,
+  NativeModules,
+  StyleSheet,
+  Platform,
+  ScrollView,
+  Text,
+  View,
+} from 'react-native';
+import { Constants } from 'expo';
+import jasmineModule from 'jasmine-core/lib/jasmine-core/jasmine';
+import Immutable from 'immutable';
+
+const { ExponentTest } = NativeModules;
+
+export default class TestScreen extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = TestScreen.initialState;
+    this._results = '';
+    this._failures = '';
+    this._scrollViewRef = null;
+
+    this._runTests = this._runTests.bind(this);
+  }
+
+  componentDidMount() {
+    const { navigation } = this.props;
+    const selectedModules = navigation.getParam('selected');
+    this._runTests(selectedModules);
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  static navigationOptions = {
+    title: 'Test Runner',
+  };
+
+  static initialState = {
+    portalChildShouldBeVisible: false,
+    state: Immutable.fromJS({
+      suites: [],
+      path: ['suites'], // Path to current 'children' List in state
+    }),
+    done: false,
+  };
+
+  setPortalChild = testPortal => {
+    if (this._isMounted) return this.setState({ testPortal });
+  };
+
+  cleanupPortal = () => {
+    return new Promise(resolve => {
+      if (this._isMounted) this.setState({ testPortal: null }, resolve);
+    });
+  };
+
+  async _runTests(modules) {
```

Comment:
replace with `_runTests = async modules => {`
then you can get rid of the `bind` call in the constructor